deploy/setup.sh

```diff
@@ -0,0 +1,280 @@
+        echo "Could not install dependancy software for guacamole, pls install guacamole manually"
+        exit
+    fi
+    echo "installing libvncserver-dev libpulse-dev libwebp-dev libssl-dev libvorbis-dev"
+    result=$(sudo apt-get install -y libvncserver-dev libpulse-dev libwebp-dev libssl-dev libvorbis-dev)
+    if grep -q "Unable to lacate" <<< $result; then
+        echo "Could not install dependancy software for guacamole, pls install guacamole manually"
+        exit
+    fi
+}
+
+install_and_config_guacamole() {
+    result=$(sudo service guacd restart)
+    if grep -q "SUCCESS" <<< $result; then
+        echo "guacamole is installed!"
+        return
+    fi
+    echo "installing guacamole"
+    # install and Configure guacamole
+    cd /home/opentech && rm -f download && rm -rf guacamole-server-0.9.9
+    wget http://sourceforge.net/projects/guacamole/files/current/source/guacamole-server-0.9.9.tar.gz/download
+    rm -rf guacamole-server-0.9.9
+    mv download guacamole-server-0.9.9.tar.gz &&  tar -xzf guacamole-server-0.9.9.tar.gz && cd guacamole-server-0.9.9/
+    result=$(autoreconf -fi)
+    result=$(./configure --with-init-dir=/etc/init.d)
+    result=$(make clean)
+    result=$(make)
+    result=$(make install)
+    ldconfig
+    # configure guacamole client
+    wget http://sourceforge.net/projects/guacamole/files/current/binary/guacamole-0.9.9.war/download
+    mv download /var/lib/tomcat7/webapps/guacamole.war
+    # configure guacamole authentication provider
+    mkdir /usr/share/tomcat7/.guacamole
+    mkdir /etc/guacamole
+    cd /home/opentech/open-hackathon/deploy/guacamole
+    cp guacamole-sample.properties /etc/guacamole/guacamole.properties
+    cp *.jar /etc/guacamole
+    ln -s /etc/guacamole/guacamole.properties /usr/share/tomcat7/.guacamole/guacamole.properties
+    result=$(sudo service guacd restart)
+    if ! (grep -q "SUCCESS" <<< $result); then
+        echo "Fail to install guacamole, please run this script once again!"
+        exit
+    fi
+    echo "guacamole installed successfully"
+}
+
+
+install_and_config_docker() {
+    # install docker
+    result=$(sudo apt-get update)
+    if grep -q "Could not resolve" <<< $result; then
+        echo "Could not update apt-get, please solve it"
+        exit
+    fi
+    result=$(sudo apt-get install apt-transport-https ca-certificates)
+    if grep -q "Unable to lacate" <<< $result; then
+        echo "Could not install dependancy software for docker, pls install docker manually"
+        exit
+    fi
+
+    result=$(sudo apt-get update)
+    if grep -q "Could not resolve" <<< $result; then
+        echo "Could not update apt-get, please solve it"
+        exit
+    fi
+    result=$(sudo apt-get purge lxc-docker)
+    if grep -q "Unable to lacate" <<< $result; then
+        echo "Could not install docker, pls install docker manually"
+        exit
+    fi
+    result=$(sudo apt-cache policy docker-engine)
+    # for ubuntu 15
```

Comment:
Do we need to check the ubuntu version first? Be curious what happens if I run the script in Ubuntu 12.04?